### ISSUE TITLE
Buggy collapsing in PHP containing HTML in-between

### ISSUE THREAD

**Issue by nikunjbhatt** [notepad-plus-plus/notepad-plus-plus#9054]
### Description of the issue
If there is HTML between 2 PHP zones and a control structure like `if`, `switch`, `for` etc is going through these zones then on collapsing the control structure in first zone is collapsing contents in the first zone only instead of collapsing up to its end bracket.

### Steps to Reproduce the Issue
1. Write following in a PHP file (attached first screenshot):
```php
<?php
if(1) {
	// some php code
?>
	<p>some html</p>
<?php
	// some php code
}
```
2. Collapse the `if` line (line # 2)
3. The content up to the 4th line (inclusive) (only that PHP zone) is collapsing 
 
### Expected Behavior
It should collapse to the ending bracket of `if` and show as following:
```php
<?php
if(1) {
```

### Actual Behavior (attached 2nd screenshot)
```php
<?php
if(1) {
	<p>some html</p>
<?php
	// some php code
}
```

### Debug Information
Notepad++ v7.9   (64-bit)
Build time : Sep 22 2020 - 03:19:04
Path : C:\Program Files\Notepad++\notepad++.exe
Admin mode : ON
Local Conf mode : OFF
OS Name : Microsoft Windows 7 (64-bit) 
OS Build : 7601.0
Current ANSI codepage : 1252
WINE : 5.19
Plugins : BetterMultiSelection.dll ComparePlugin.dll DSpellCheck.dll Explorer.dll mimeTools.dll NppConverter.dll NppExport.dll zoomdisabler_x64.dll 

**Screenshot 1**
![Selection_028](https://user-images.githubusercontent.com/4369071/97067060-77621280-15d7-11eb-9b3a-c60bf6ff7f8e.png)

**Screenshot 2**
![Selection_029](https://user-images.githubusercontent.com/4369071/97067061-792bd600-15d7-11eb-9e6c-eb0bdb69cfb0.png)

**Comment by ArkadiuszMichalski** [comment]
@nikunjbhatt 
This is a problem with Scintilla, not Notepad++. You can report this bug to Scintilla (for `LexHTML.cxx`): https://sourceforge.net/p/scintilla/bugs/

Although, even in Visual Studio Code (or Sublime Text), folding a mixed-up `if` against PHP zones doesn't work. But such code is still correct and works without errors:
```php
<?php
if(1) {
	echo "a";
?>
	<p>some html</p>
<?php
	echo "b";
}
echo "\nc";
```

**Comment by nikunjbhatt** [comment]
Here is the link to the bug reported in Scintilla: https://sourceforge.net/p/scintilla/bugs/2220/

**Comment by allenmoorestl** [comment]
I had a similar post.  Not sure if it changes anything that the curly braces do accurately "find" the start/end pairs correctly (i.e. when you place your cursor next to the brace at the end of the code structure (line 8 above) it correctly turns that one and the one in line 2 red indicating that they are pairs.

**Comment by nikunjbhatt** [comment]
@allenmoorestl Right, that's the accurate point - the collapsing behaviour is different than the matching of bracket pairs.

See this example:
![Selection_037](https://user-images.githubusercontent.com/4369071/101059567-6df39f00-35b4-11eb-936d-e8c4bef51b05.png)

According to the current implementation, only the 3 to 5 lines should get hidden, but what actually happening is this:
![Selection_038](https://user-images.githubusercontent.com/4369071/101059787-a98e6900-35b4-11eb-951f-c5ea56412f16.png)

... and this is correct, logical.
